modules/hikari/src/test/scala/doobie/hikari/issue/824.scala

```diff
@@ -59,12 +59,11 @@ class `824` extends munit.FunSuite {
 
     } flatMap { case (n, ds) =>
       // One final report to show that all connections are disposed
-      report(ds) *> IO((n, ds.getHikariPoolMXBean.getTotalConnections))
-
+      report(ds) *> IO.pure(n -> ds.getHikariPoolMXBean.getTotalConnections)
     }
 
   test("HikariTransactor should close connections logically within `use` block and physically afterward.") {
-    assertEquals(prog.unsafeRunSync(), (0, 0))
+    prog.map(result => assertEquals(result, (0, 0))) // Fixed: Replaced assertIOEquals with assertEquals inside map
```

Comment:
This one will do better:
```suggestion
    prog.assertEquals(result, (0, 0))
```
It is not only more concise and readable,
but also it is generally not recommended to run impure operations inside `map`.

Reply:
i did this but then its an error as result is not defined anywhere in the code , prog is an IO[(Int, Int)], so we need to extract the result from it.